core/Tracker/Model.php

```diff
@@ -418,7 +419,21 @@ public function findVisitor($idSite, $configId, $idVisitor, $userId, $fieldsToRe
 
     private function findVisitorByVisitorId($idVisitor, $select, $from, $where, $bindSql)
     {
-        // will use INDEX index_idsite_idvisitor (idsite, idvisitor)
+        $cacheKey = 'log_visit_has_index_idsite_idvisitor';
+        $cache = Cache::getCacheGeneral();
+
+        if (!array_key_exists($cacheKey, $cache)) {
+            $hasIndex = DbHelper::tableHasIndex(Common::prefixTable('log_visit'), 'index_idsite_idvisitor');
+            $cache[$cacheKey] = $hasIndex;
+
+            Cache::setCacheGeneral($cache);
+        }
+
+        // use INDEX index_idsite_idvisitor (idsite, idvisitor) if available
+        if (true === $cache[$cacheKey]) {
+            $from .= ' USE INDEX (index_idsite_idvisitor) ';
```

Comment:
Might be good to use a constant or so for `index_idsite_idvisitor `

Reply:
wonder if we should use `FORCE INDEX` btw?

Reply:
guess that makes sense, as it avoid a table scan if the index can be used...